src/backend/tests/it/utils/assertion.rs

```diff
@@ -19,6 +20,25 @@ pub fn assert_custom_tokens_eq(
     assert_eq!(results_tokens.len(), expected_tokens.len());
 
     for (token, expected) in results_tokens.iter().zip(expected_tokens.iter()) {
-        assert!(expected == token);
+        assert!(expected.token == token.token);
+        assert_eq!(expected.enabled, token.enabled);
+    }
+}
+
+pub fn assert_some_tokens_timestamp<T>(tokens: Vec<T>)
+where
+    T: TokenTimestamp,
+{
+    for token in tokens.iter() {
+        assert!(token.get_timestamp().is_some());
+    }
+}
+
+pub fn assert_none_tokens_timestamp<T>(tokens: Vec<T>)
+where
+    T: TokenTimestamp,
+{
+    for token in tokens.iter() {
+        assert!(token.get_timestamp().is_none());
```

Comment:
Likewise, I would recommend adding a message to the assertion to say which token is flawed.

Reply:
Fun fact, I explicitely did not added stacktrace in the test. No particular reason. Added now.